src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ExpressionBuilder.java

```diff
@@ -115,6 +125,14 @@ private static Expression<BuiltCondition> buildExpressionRecursive(
     if (conditionExpressions.isEmpty()) {
       return null;
     }
+    // when having an empty array $in, if $in occurs within an $and logic, entire $and should match
+    // nothing
+    if (hasInFilterThisLevel
```

Comment:
Instead of these state flags, is it not possible to translate empty `$in` clause into something in CQL that evaluates to `false`? Instead of trying to use special states, checks, at this level. I am mostly worried that due to possible nesting, these checks might get complicated and not handle all cases.

Reply:
same concerns, but it seems there is no keywords like 'false' you can add into cql where clause

Reply:
One thing that I think does work is actually using dummy `ID` clause for our own `key` column:

```
   WHERE key in ( )
```

which works as expected, matching nothing. And can then be composed with other clauses, like

```
   WHERE key in ( ) OR array_contains contains "column value"
```

(or however it would be generated)

Reply:
unfortunately, this is also a no.
You can not use OR for partition key in cassandra.
<img width="1521" alt="image" src="https://github.com/stargate/jsonapi/assets/138714577/fe66faa9-262a-43ad-bc04-e8f36e291b8e">

Reply:
Interesting. Ok, never mind then... Cassandra not smart enough to optimize that away (or validates usage before optimizations).

Let's go with code as is, then.